src/server/command_registry.h

```diff
@@ -169,8 +169,16 @@ class CommandRegistry {
   CommandRegistry& operator<<(CommandId cmd);
 
   const CommandId* Find(std::string_view cmd) const {
-    auto it = cmd_map_.find(cmd);
-    return it == cmd_map_.end() ? nullptr : &it->second;
+    if (const auto it = cmd_map_.find(cmd); it != cmd_map_.end()) {
+      return &it->second;
+    }
+
+    if (const auto it = cmd_aliases_.find(cmd); it != cmd_aliases_.end()) {
+      if (const auto alias_lookup = cmd_map_.find(it->second); alias_lookup != cmd_map_.end()) {
```

Comment:
Shouldn't we expect that `it->second` will exist in the `cmd_map_`? In other words shouldn't this if be a `DCHECK + a find`. The registry is not mutable and the alias map is initialized during bootstrap so I would expect that this `alias_lookup == cmd_map_.end()` can't be true right ? 

Basically what I am trying to say is that the alias map should never point to a command that does not exist in the `cmd_map` correct?

Reply:
The aliases are processed before commands are registered, in the ctor of command registry. It wouldn't be possible to validate if the alias is correct at that time.